src/roles/factotum.js

```diff
@@ -91,6 +100,156 @@ class Factotum extends MetaRole {
         }
       }
     }
+
+    creep.memory.labs = 'fill'
+  }
+
+  fillFeeders (creep) {
+    const reaction = creep.room.getActiveReaction()
+    const feeders = creep.room.getFeederLabs()
+    if (feeders.length < 2) {
+      delete creep.memory.labs
+      delete creep.memory.filling
+      return
+    }
+
+    if (feeders[0].mineralAmount > 0 && feeders[0].mineralType !== reaction[0]) {
+      creep.memory.labs = 'empty'
+      return
+    }
+    if (feeders[1].mineralAmount > 0 && feeders[1].mineralType !== reaction[0]) {
+      creep.memory.labs = 'empty'
+      return
+    }
+
+    if (_.sum(creep.carry) <= 0) {
+      if (feeders[0].mineralAmount / feeders[0].mineralCapacity >= 0.5) {
+        if (feeders[0].mineralAmount / feeders[0].mineralCapacity >= 0.5) {
```

Comment:
This line looks like it should be `        if (feeders[1].mineralAmount / feeders[1].mineralCapacity >= 0.5) {`.